### DISCUSSION TITLE
Prism 8.1 questions

### DISCUSSION THREAD

**Discussion by dzebas** [PrismLibrary/Prism#2548] [Xamarin.Forms] [answered]
Hi,

At the moment we are using Prism 7.2 in our enterprise app (hasn't been released yet). We have complex objects which we present in ContentPage with DevExpress TabView control to separate those big objects into partial views. We load partials views with ViewModelLocator.AutowirePartialView and all works really well, partial views gets parameters from main view model etc.

Now we're trying to adapt our app to tablets, so we want to use FlyoutPage (aka MasterDetailPage), so customers can select object from the list in master view and display object data in detail view.

I've couple of questions:

- Because AutowirePartialView is no longer available in Prism 8.1 do we need to use regions now? Are regions compatible with third party TabView control? I tried following some online demos (not much info) and was not able to pass parameters to a region [Stackoverflow](https://stackoverflow.com/questions/69082546/xamarin-forms-prism-regions-in-tabview)

- Regarding FlyoutPage - can we present it modally? Is there any examples how to use it with Prism?

- If I understand correctly, to get help with Prism 8 we need to become sponsors? I don't have a problem with this and can ask company to pay if we get some form of support.

Thanks in advance.

**Comment by dansiegel** [comment, marked as answer]
@dzebas a little out of order but hope this helps answer some questions

> If I understand correctly, to get help with Prism 8 we need to become sponsors? I don't have a problem with this and can ask company to pay if we get some form of support.

Being a $10/month sponsor is far from reasonable for support. As is noted, if your organization requires actual Enterprise level support that can be done through my consulting company [AvantiPoint](https://avantipoint.com/contact). If your company doesn't really require an Enterprise Support contract but would like to step up and help make Prism a sustainable OSS project, I do offer 2 Corporate level sponsorships which do actually come with VIP Support each month as well (note this works particularly well if you're sponsoring from a GitHub organization). All of our sponsors at the Supporter level or above do get access to Sponsor Connect which includes a private NuGet Feed and a Sponsor only Discord Channel. 

> Regarding FlyoutPage - can we present it modally? Is there any examples how to use it with Prism?

Prism can do anything that you can do in Xamarin.Forms. Remember that under the covers we're just invoking the Xamarin.Forms API's for you. ANY page can be made modal in Prism.Forms by simply adding the `useModalNavigation` query parameter like: `SomePage?useModalNavigation=true/AnotherPage/SomeOtherModal?useModalNavigation=true`. I'm not sure what you're looking for on this. If the question is "Do we have an example using the FlyoutPage?" Then the answer is no, it's a newer page type and we don't have any, and are unlikely to add any samples with it.

> Because AutowirePartialView is no longer available in Prism 8.1 do we need to use regions now? Are regions compatible with third party TabView control?

The short answer here is that yes you do need to use Regions. Regions are compatible with 3rd party controls, however you will need to create and register the Adapters for the 3rd party controls yourself.

**Reply by dzebas** [reply]
Hi @dansiegel,

Thank you for detailed explanation. I will talk to my manager re sponsorship. I think for now we will stick to Prism 7.2 and use obsolete MasterDetailPage as we are pressured by time and redoing whole app to use regions etc. is not a simple task.
